universal/include/userver/crypto/ssl_ctx.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+/// @file userver/crypto/ssl_ctx.hpp
+/// @brief @copybrief crypto::SslCtx
+
+#include <memory>
+#include <vector>
+
+#include <userver/crypto/certificate.hpp>
+#include <userver/crypto/private_key.hpp>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace crypto {
+
+/// @ingroup userver_universal userver_containers
+///
+/// SSL context
+class SslCtx {
+public:
+    static SslCtx CreateServerTlsContext(
+        const crypto::CertificatesChain& cert_chain,
+        const crypto::PrivateKey& key,
+        const std::vector<crypto::Certificate>& extra_cert_authorities = {}
+    );
+
+    static SslCtx CreateClientTlsContext(const std::string& server_name);
+
+    static SslCtx CreateClientTlsContext(
+        const std::string& server_name,
+        const crypto::Certificate& cert,
+        const crypto::PrivateKey& key,
+        const std::vector<crypto::Certificate>& extra_cert_authorities = {}
+    );
+
+    SslCtx(SslCtx&&) noexcept;
+    SslCtx& operator=(SslCtx&&) noexcept;
+    ~SslCtx();
+
+    SslCtx(const SslCtx&) = delete;
+    SslCtx& operator=(const SslCtx&) = delete;
+
+    void* GetRawSslCtx() const noexcept;
```

Comment:
What's the point of removing the type information if it's casted to a specific type when this method is called?